modular_skyrat/modules/morgue_radio/morgue.dm

```diff
@@ -0,0 +1,41 @@
+/obj/structure/bodycontainer/morgue/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/morgue_radio)
+
+/obj/structure/bodycontainer
+	/// reference to internal radio in the morgue trays.
+	var/obj/item/radio/headset/headset_med/radio
+
+/datum/component/morgue_radio
+	/// Typecasted reference to the current tray.
+	var/obj/structure/bodycontainer/morgue/morgue
+
+
+/datum/component/morgue_radio/RegisterWithParent()
+	morgue = parent
```

Comment:
You need to use checks like these on components to ensure that they don't runtime if anyone tries to add them to anything else. I _think_ this should work, but you should 100% test it to be sure.
```suggestion
	morgue = parent
	if(!istype(morgue))
		qdel(src)
		return

```

Reply:
oh it won't runtime, it will simply not work i think

Reply:
oh it won't runtime, it will simply not work i think

Reply:
must return component incompatible